README.md

```diff
@@ -1,124 +1,132 @@
+`yarn add formsy-react react react-dom` and use with webpack, browserify, etc.
 
-## Changes
+## Quick Start
 
-[Check out releases](https://github.com/christianalfoni/formsy-react/releases)
+### 1. Build a Formsy element
 
-[Older changes](CHANGES.md)
+```jsx
+// MyInput.js
+import { withFormsy } from 'formsy-react';
+import React from 'react';
 
-## How to use
+class MyInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.changeValue = this.changeValue.bind(this);
+  }
 
-See [`examples` folder](/examples) for examples. [Codepen demo](http://codepen.io/semigradsky/pen/dYYpwv?editors=001).
+  changeValue(event) {
+    // setValue() will set the value of the component, which in
+    // turn will validate it and the rest of the form
+    // Important: Don't skip this step. This pattern is required
+    // for Formsy to work.
+    this.props.setValue(event.currentTarget.value);
+  }
+
+  render() {
+    // An error message is returned only if the component is invalid
+    const errorMessage = this.props.getErrorMessage();
+
+    return (
+      <div>
+        <input
+          onChange={this.changeValue}
+          type="text"
+          value={this.props.getValue() || ''}
+        />
+        <span>{errorMessage}</span>
+      </div>
+    );
+  }
+}
+
+export default withFormsy(MyInput);
+```
 
-Complete API reference is available [here](/API.md).
+`withFormsy` is a [Higher-Order Component](https://facebook.github.io/react/docs/higher-order-components.html) that exposes additional props to `MyInput`. See the [API](/API.md#withFormsy) documentation to view a complete list of the props.
 
-#### Formsy gives you a form straight out of the box
+### 2. Use your Formsy element
 
 ```jsx
-  import Formsy from 'formsy-react';
-
-  class MyAppForm extends React.Component {
-    state = { canSubmit: false };
-    enableButton = () => {
-      this.setState({
-        canSubmit: true
-      });
-    }
-    disableButton = () => {
-      this.setState({
-        canSubmit: false
-      });
-    }
-    submit(model) {
-      someDep.saveEmail(model.email);
-    }
-    render() {
-      return (
-        <Formsy.Form onValidSubmit={this.submit} onValid={this.enableButton} onInvalid={this.disableButton}>
-          <MyOwnInput name="email" validations="isEmail" validationError="This is not a valid email" required/>
-          <button type="submit" disabled={!this.state.canSubmit}>Submit</button>
-        </Formsy.Form>
-      );
-    }
-  });
+import Formsy from 'formsy-react';
+import React from 'react';
+import MyInput from './MyInput';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.disableButton = this.disableButton.bind(this);
+    this.enableButton = this.enableButton.bind(this);
+    this.submit = this.submit.bind(this);
+    this.state = { canSubmit: false };
+  }
+
+  disableButton = () => {
```

Comment:
Use ES7 class property + arrow function (as is here), or bind in the constructor (L81)... I've preferred arrow functions so you can easily see what functions are being bound when they are declared as opposed to having to look in the constructor. 🙂